### PR TITLE
feat: Move frames context reading into contextifyFramesIntegration

### PR DESCRIPTION
**Pull request by kamilogorek**

### CODE REVIEW

File: integrations.go [hunk 1]
```diff
@@ -260,3 +260,104 @@ func getIgnoreErrorsSuspects(event *Event) []string {
...
+		components = components[1:]
+		possibleLocation := strings.Join(components, "/")
+
+		if fileExists(possibleLocation) {
+			cfi.cachedLocations[originalPath] = possibleLocation
+			return possibleLocation
+		}
+	}
+
+	return ""
```
Comment (bruno-garcia) [inline integrations.go]: Will lookup forever in this case. Needs caching result

File: integrations.go [hunk 2]
```diff
@@ -260,3 +260,104 @@ func getIgnoreErrorsSuspects(event *Event) []string {
...
+
+	for _, frame := range frames {
+		if !frame.InApp {
+			contextifiedFrames = append(contextifiedFrames, frame)
+			continue
+		}
+
+		var path string
+		// Optimize for happy path here
+		if fileExists(frame.AbsPath) {
```
Comment (bruno-garcia) [inline integrations.go]: Lookup the cache first to avoid I/O

File: integrations.go [hunk 3]
```diff
@@ -260,3 +260,104 @@ func getIgnoreErrorsSuspects(event *Event) []string {
...
+		} else {
+			path = cfi.findNearbySourceCodeLocation(frame.AbsPath)
+		}
+
+		if path == "" {
+			contextifiedFrames = append(contextifiedFrames, frame)
+			continue
+		}
+
+		lines, initial := cfi.sr.readContextLines(path, frame.Lineno, cfi.contextLines)
```
Comment (bruno-garcia) [inline integrations.go]: Blocking I/O?

File: integrations.go [hunk 4]
```diff
@@ -260,3 +260,104 @@ func getIgnoreErrorsSuspects(event *Event) []string {
...
+		if fileExists(possibleLocation) {
+			cfi.cachedLocations[originalPath] = possibleLocation
+			return possibleLocation
+		}
+	}
+
+	return ""
+}
+
+func (cfi *contextifyFramesIntegration) addContextLinesToFrame(frame Frame, lines [][]byte, initial int) Frame {
```
Comment (bruno-garcia) [inline integrations.go]: `initial` should be called `context_line` or something more meaningful?

File: integrations.go (hunk 3 above)
Comment (kamilogorek) [reply integrations.go]: Yes, but it's all cached in the source reader once done

File: integrations.go (hunk 2 above)
Comment (kamilogorek) [reply integrations.go]: Fixed

File: integrations.go (hunk 3 above)
Comment (bruno-garcia) [reply integrations.go]: When it comes to blocking I/O, there's no `yes, but` 👿 
I guess ASP.NET Core is a bad influence?! or is it?

**Pull request merged**